Chap_Terms.tex

```diff
@@ -4,39 +4,30 @@
+The \ac{PMIx} \ac{ASC} has further adopted a policy that modification of existing released \acp{API} will only be permitted under extreme circumstances. In its effort to avoid introduction of any such backward incompatibility, the community has avoided the definitions of large numbers of \acp{API} that each focus on a narrow scope of functionality, and instead relied on the definition of fewer generic \acp{API} that include arrays of key-value attributes for ``tuning'' the function's behavior. Thus, modifications to the PMIx standard increasingly consist of the definition of new attributes along with a description of the \acp{API} to which they relate and the expected behavior when used with those \acp{API}.
 
+The following terms and terminology is used throughout this document:
 \begin{itemize}
-\item \declareterm{session}\emph{session} refers to an allocated set of resources assigned to a particular user by the system \ac{WLM}. Historically, \ac{HPC} sessions have consisted of a static allocation of resources - i.e., a block of resources are assigned to a user in response to a specific request and managed as a unified collection. However, this is changing in response to the growing use of dynamic programming models that require on-the-fly allocation and release of system resources. Accordingly, the term \emph{session} in this document refers to the current block of assigned resources and is a potentially dynamic entity.
-\item \declareterm{slot}\declareterm{slots}\emph{slot} refers to an allocated entry for a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
-\item \declareterm{job}\emph{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a session. For example, ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' is considered a single \ac{MPMD} job containing two applications.
+\item \declareterm{session}\emph{session} refers to an allocated set of resources assigned to a particular user by the system \ac{WLM}. Historically, \ac{HPC} \emph{sessions} have consisted of a static allocation of resources - i.e., a block of resources are assigned to a user in response to a specific request and managed as a unified collection. However, this is changing in response to the growing use of dynamic programming models that require on-the-fly allocation and release of system resources. Accordingly, the term \emph{session} in this document refers to the current block of assigned resources and is a potentially dynamic entity.
```

Comment:
It's odd if we look at the allocation as independent entities. In which case the RM could create a unique session for reach WLM allocation. However, for a dynamic environment where the user requests their allocation grow or shrink (possibly with the WLM creating an allocation for the new addition, but making it depended on the requesting allocation) we needed wording to capture that.

We're open to suggestions on how to better capture that.

Reply:
I think what Martin's getting at is not the dynamic increase/decrease of resources in a set, but completely independent allocations that just happen to be concurrent. 

If I am right about what Martin means, perhaps rewording slightly: 'session refers to a set of resources allocated as a logical unit and assigned to a particular user by the system WLM, sometimes referred to as a "job" or "allocation".'

Reply:
What is the difference to a job then? Should we just drop "Session"? What is the difference between a session, a job, and an application? What is the intention if a user submits two allocations - are those one session or two? I am not suggestion a change, I am just trying to understand the intention behind these terms.

Reply:
Oh, oops. I shouldn't have suggested the word "job" as a synonym since in the rest of the document a "job" refers to the execution of applications as a single unit (aka a "job step" in SLURM). I was thinking colloquially and not how we have things defined in the document.

I do see the word "allocation" seemingly used interchangeably with session in the document. Sometimes it is used as part of the definition of a SESSION attribute, e.g. 3.2.11 ("Data Persistence Structure")

Possibly confusingly 3.2.20  and 3.4.27 (I think that is right, the numbering is messed up in my build) the text talks about allocations with respect to jobs. In these cases (and there may be others in the doc), I think "jobs" should be changed to "sessions" since in our definition "sessions" refers to allocated resources while a "job" does not. I don't suggest that as part of this PR, but if indeed we mean "sessions" where it says "jobs" in these cases, we should fix that in a new PR

Reply:
application:  A collection of processes that share the same expression of executable, environment and other associated parameters within a job.

Reply:
Todo:  Let's do a review of use in the standard and make sure this definition is compatible with current usage

Reply:
I will post a PR soon with this text, though I do find it is unfortunate, that the term application does appear to be used in multiple ways in the document.  At times it is used very generically to refer to the launching of possibly multiple job steps that coordinate to solve a problem and at other times it is used in this very narrow context to refer to the expression of an executable.

Reply:
So why not just broaden the definition to be: generically to refer to the launching of possibly multiple job steps that coordinate to solve a problem, or  refer to the expression of an executable? The context makes it clear which definition is being referenced. No need to overthink this 😄